### PR TITLE
chore(docs): update docs for renku-python 1.0.0 release

### PR DESCRIPTION
**Pull request by Panaetius**
closes https://github.com/SwissDataScienceCenter/renku-python/issues/2400

### CODE REVIEW

**Comment by rokroskar** [comment]
@Panaetius should we merge this once the docs reorg is merged and once renku-python 1.0 is out?

**Comment by Panaetius** [comment]
@rokroskar would there be an issue with merging it before renku-python 1.0.0? if not, i'd say merge it

**Comment by rokroskar** [comment]
Not sure tbh - though all of the links currently live on renkulab point to the latest docs so users will see this which I guess would be confusing.

File: docs/topic-guides/provenance.rst
Comment (vigsterkr) [inline docs/topic-guides/provenance.rst:110]:
```suggestion
   set to the name specified by ``--name``. This is a workflow specification
```

File: docs/topic-guides/provenance.rst
Comment (vigsterkr) [inline docs/topic-guides/provenance.rst:138]:
```suggestion
**renku workflow execute**. For understanding the basic functionality, head to
```

**Review by vigsterkr — COMMENTED** [review]
so very minor comments

**Comment by rokroskar** [comment]
I tried to rebase this and it was a mess - so I made a new branch `rp-2400-updates-for-1.0.0-2` that is based on current master and just cherry-picks the relevant commits. Feel free to change to that branch to make this PR a bit easier to parse. OTOH maybe github will just do the right thing when it's merged?

**Review by rokroskar — APPROVED** [review]
🚀

**Pull request merged**
